### ISSUE TITLE
*.json files with no `validKeys` blow up `TypeError: Cannot read property '0' of null` in ExportDefaultDeclaration.prototype.render

### ISSUE THREAD

**Issue by btakita** [rollup/rollup#1146]
I'm packaging an import from mime-db, which loads a json file.

https://github.com/jshttp/mime-db/blob/master/db.json

<details>
  <summary>Stack trace</summary>

Cannot read property '0' of null
TypeError: Cannot read property '0' of null
    at ExportDefaultDeclaration.render (/home/user/project/node_modules/rollup/src/ast/nodes/ExportDefaultDeclaration.js:57:4)
    at Module.render (/home/user/project/node_modules/rollup/src/Module.js:320:8)
    at /home/user/project/node_modules/rollup/src/Bundle.js:376:17
    at Array.forEach (native)
    at Bundle.render (/home/user/project/node_modules/rollup/src/Bundle.js:375:16)
    at generate (/home/user/project/node_modules/rollup/src/rollup.js:75:30)
    at Object.result.write (/home/user/project/node_modules/rollup/src/rollup.js:103:9)
    at /home/user/project/node_modules/rollup-watch/dist/rollup-watch.cjs.js:162:21
Type rollup --help for help, or visit https://github.com/rollup/rollup/wiki
</details>

See https://github.com/rollup/rollup-plugin-json/issues/19

**Comment by kzc** [comment]
When I changed ExportDefaultDeclaration I assumed that acorn would produce correct information for the AST.

Could you please tell me if this rollup patch fixes the problem for you?

```patch
--- a/src/ast/nodes/ExportDefaultDeclaration.js
+++ b/src/ast/nodes/ExportDefaultDeclaration.js
@@ -53,7 +53,7 @@ export default class ExportDefaultDeclaration extends Node {
                let declaration_start;
                if ( this.declaration ) {
                        const statementStr = code.original.slice( this.start, this.end );
-                       declaration_start = this.start + statementStr.match(/^export\s+default\s*/)[0].length;
+                       declaration_start = this.start + statementStr.match(/\s*export\s+default\s*/)[0].length;
                }
 
                if ( this.shouldInclude || this.declaration.activated ) {
```
If it doesn't work could you please post the value of `JSON.stringify(statementStr)`?

**Comment by kzc** [comment]
Likely fix: #1152

**Comment by Rich-Harris** [comment]
Fixed on both ends now, so I'll close this

**Comment by yairEO** [comment]
I'm a regular user, using Rollup for over a year with Gulp. I'm getting this error even after updating to v0.40.0:


````
[14:44:16] Rollup [catch]:  TypeError: Cannot read property '0' of null
    at ExportDefaultDeclaration.render (C:\www\UI\BUILD\node_modules\rollup\dist\rollup.js:6225:82)
    at Module.render (C:\www\UI\BUILD\node_modules\rollup\dist\rollup.js:7767:8)
    at C:\www\UI\BUILD\node_modules\rollup\dist\rollup.js:9080:23
    at Array.forEach (native)
    at Bundle.render (C:\www\UI\BUILD\node_modules\rollup\dist\rollup.js:9079:22)
    at Object.generate (C:\www\UI\BUILD\node_modules\rollup\dist\rollup.js:9316:26)
    at C:\www\UI\BUILD\gulpfile.js:657:33
    at lib$es6$promise$$internal$$tryCatch (C:\www\UI\BUILD\node_modules\es6-promise\dist\es6-promise.js:362:16)
    at lib$es6$promise$$internal$$invokeCallback (C:\www\UI\BUILD\node_modules\es6-promise\dist\es6-promise.js:374:17)
    at lib$es6$promise$$internal$$publish (C:\www\UI\BUILD\node_modules\es6-promise\dist\es6-promise.js:345:11)
    at lib$es6$promise$asap$$flush (C:\www\UI\BUILD\node_modules\es6-promise\dist\es6-promise.js:119:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickCallback (node.js:401:11)
````

**Comment by kzc** [comment]
Fixed in 0.40.1

Allow missing space between export default and declaration (#1218)
